app/imageUtils.js

```diff
@@ -0,0 +1,208 @@
+        const hasPerms = await hasIOPermissions(dirPath)
+        if(!hasPerms) throw new Error(`Invalid R/W Permissions at ${dirPath}`)
+        
+        const filePath = path.join(dirPath, `${imageName}.${imageFormat}`)
+        if(!fs.existsSync(filePath)) throw new Error(`Error: ${filePath} does not exist`)
+        const imageBuffer = await fs.promises.readFile(filePath)
+        res.set('Content-Type', mimeType)
+        return res.send(imageBuffer)
+    } catch(err) {
+        logger.info(`__getImage: ${err}`)
+        return res.sendStatus(400)
+    }
+}
+
+async function getArticleImage(res, imageName, webpFormat, width) {
+    try {
+        if(!imageName) throw new Error('GetArticleImages: Invalid Parameters', imageName)
+        return await __getImage(res, imageName, ARTICLE, webpFormat, width)
+    } catch(err) {
+        logger.info(`GetArticleImage ${err}`)
+        return res.sendStatus(500)
+    }
+}
+
+async function setArticleContentImage(imageData) {
+    try {
+        if(!imageData) throw new Error('setContentImage: Invalid Parameters', imageData)
+        const imageName = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 10).concat(`.${JPEG}`)
+        const hasBeenSaved = await __saveImage(imageData, imageName, ARTICLE, CONTENT_IMAGE_SIZES)
+        if(hasBeenSaved) return imageName
+        throw new Error('SetArticleContentImage: Couldn\'t Save Image')
+    } catch(err) {
+        return logger.info(`SetArticleContentImage: ${err}`)
+    }
+}
+
+async function setArticleHeaderImage(headerData, linkId) {
+    try {
+        if(!headerData) throw new Error('setContentImage: Invalid Parameters', headerData)
+        const article = await Article.findOne({link: linkId}).exec()
+        article.headerUrl = createRandomString(ARTICLE_HEADER_IMAGENAME_LENGTH)
+        article.save()
+        return await __saveImage(headerData, article.headerUrl, ARTICLE, HEADER_IMAGE_SIZES)
+    } catch(err) {
+        return logger.info(`SetArticleHeaderImage: ${err}`)
+    }
+}
+
+async function getProfileImage(res, imageName, format, width) {
+    try {
+        if(!imageName) throw new Error('getProfileImage Error: Invalid imageName: ', imageName)
+        const user = await User.findOne({link: imageName}).exec()
+        return __getImage(res, user.avatar, PROFILE, format, width)
+    } catch(err) {
+        logger.info('getProfileImage Error:' + err)
+        return res.sendStatus(500)
+    }
+}
+
+async function setProfileImage(link, image) {
+    try {
+        if(!link) throw new Error('getProfileImage Error: Invalid Param: ', link)
+        const imageName = createRandomString(USER_PROFILE_IMAGENAME_LENGTH)
+        const user = await User.findOne({link}).exec()
+        user.avatar = imageName
+        user.save()
+        return await __saveImage(image, imageName, PROFILE, PROFILE_IMAGE_SIZES)
+    } catch(err) {
+        logger.info('setProfileImage Error:' + err)
```

Comment:
won't it display stack trace, logging out the function name might be unecessary

Reply:
Not all the time, it makes debugging significantly easier just to add this to the catch logs